src/bylines/index.js

```diff
@@ -3,64 +3,520 @@
+
+		for ( const mutation of mutationList ) {
+			if ( mutationTypes.includes( mutation.type ) ) {
+				let rootElement;
+
+				// If mutation.type is characterData, target the outer span element.
+				// Otherwise, on childList and subtree, remove the target itself.
+				if ( mutation.type === 'characterData' ) {
+					rootElement = mutation.target.parentNode.parentNode ?? null;
+				} else {
+					rootElement = mutation.target ?? null;
+				}
+
+				// Remove token element.
+				if (
+					rootElement &&
+					rootElement?.matches?.( '.token-inline-block' )
+				) {
+					rootElement.remove();
+				}
+
+				// Update byline meta.
+				updateBylineMeta( bylineElement );
+
+				queryTokensInUse( bylineElement );
+			}
+		}
+	};
+
+	/**
+	 * Add a Mutation Observer to byline element, so when a token is edited trough delete/backspace it
+	 * gets removed as it was clicked.
+	 *
+	 * @param {Element} bylineElement
+	 */
+	const addMutationObserverToByline = bylineElement => {
+		// Mutation observer config.
+		const config = {
+			childList: true,
+			subtree: true,
+			characterData: true,
+			characterDataOldValue: true,
+		};
+
+		const observer = new MutationObserver( handleMutation );
+
+		// Start observing the target node for configured mutations
+		observer.observe( bylineElement, config );
+	};
+
+	/**
+	 * Handle Error
+	 *
+	 * @param {Error} error
+	 */
+	function handleError( error ) {
+		if ( 'AbortError' === error.name ) {
+			return;
+		}
+		noticesDispatch.createErrorNotice( error.message, {
+			isDismissible: true,
+		} );
+	}
+
+	/**
+	 * Enabled toggle handler.
+	 *
+	 * @param {boolean} value Boolean, true if custom byline is enabled, false if not.
+	 */
 	const handleEnableToggle = value => {
 		editPost( { meta: { [ newspackBylines.metaKeyActive ]: value } } );
 		setIsEnabled( value );
-	}
-	// Byline change handler.
-	const handleBylineChange = value => {
-		const tags = value.match( /<[^>]+>/g );
-		if ( tags && tags.some( tag => ! tag.startsWith( '<Author' ) && ! tag.startsWith( '</Author' ) ) ) {
-			alert( __( 'Only the <Author> tag is allowed.', 'newspack-plugin' ) ); // eslint-disable-line no-alert
+	};
+
+	/**
+	 * Add event listener for token removal on document, since the tokens are dynamically
+	 * inserted into byline element.
+	 */
+	useEffect( () => {
+		documentRef.current.addEventListener( 'click', function ( event ) {
+			// Check if clicked element is token remove button.
+			if (
+				event.target.classList.contains( 'token-inline-block__remove' )
+			) {
+				const bylineElement = document.querySelector(
+					'.newspack-byline-textarea'
+				);
+
+				if (
+					bylineElement.querySelector(
+						'span#token-' + event.target.dataset.token
+					)
+				) {
+					// Remove token element.
+					bylineElement
+						.querySelector(
+							'span#token-' + event.target.dataset.token
+						)
+						.remove();
+
+					// Update byline meta.
+					updateBylineMeta( bylineElement );
```

Comment:
Does this need to be called directly? I think removing the token from the field should trigger the mutation observer.

Reply:
You're right, they don't seen to be necessary there, the mutation observer will be already triggered and it will call it